test/functional/aggregation_tests.js

```diff
@@ -1,7 +1,168 @@
 'use strict';
 var expect = require('chai').expect;
+const ReadPreference = require('mongodb-core').ReadPreference;
+const sinon = require('sinon');
 
 describe('Aggregation', function() {
+  it('Should use secondary readPreference if given', {
+    metadata: { requires: { topology: 'replicaset' } },
+
+    test: function(done) {
+      const configuration = this.configuration;
+
+      const client = configuration.newClient(
+        { w: 1, readConcern: { level: 'majority' } },
+        { poolSize: 1 }
+      );
+
+      // Trigger test once whole set is up
+      client.connect((err, client) => {
+        expect(err).to.be.null;
+
+        const db = client.db(this.configuration.db);
+        // Some docs for insertion
+        const docs = [
+          {
+            title: 'this is my title',
+            author: 'bob',
+            posted: new Date(),
+            pageViews: 5,
+            tags: ['fun', 'good', 'fun'],
+            other: { foo: 5 },
+            comments: [
+              { author: 'joe', text: 'this is cool' },
+              { author: 'sam', text: 'this is bad' }
+            ]
+          }
+        ];
+
+        // Create a collection
+        const collection = db.collection(
+          'shouldCorrectlyExecuteSimpleAggregationPipelineWithSecondaryReadPreference'
+        );
+        // Insert the docs
+        collection.insert(docs, { w: 1 }, (err, result) => {
+          expect(result).to.exist;
+          expect(err).to.be.null;
+
+          const internalClientCursor = sinon.stub(client.topology.s.coreTopology, 'cursor');
+          const expectedReadPreference = new ReadPreference(ReadPreference.SECONDARY);
+
+          // Execute aggregate, notice the pipeline is expressed as an Array
+          collection.aggregate(
+            [
```

Comment:
do we need to use such a big pipeline for this, where we're testing just whether a readPreference makes it to the command? It seems like you could pass an empty array and save a lot of visual space

Reply:
Alternatively, you could do what the change stream tests do, and [make the pipeline a const outside the scope of the function](https://github.com/mongodb/node-mongodb-native/blob/master/test/functional/change_stream_tests.js#L15)